ipatests/test_integration/test_installation.py

```diff
@@ -357,6 +358,53 @@ def test_ipa_ca_crt_permissions(self):
         assert owner == "root"
         assert group == "root"
 
+    def test_is_ipa_configured(self):
+        """Verify that the old and new methods of is_ipa_installed works
+
+           If there is an installation section then it is the status.
+
+           If not then it will fall back to looking for configured
+           services and files and use that for determination.
+        """
+        cmd = ['python3', '-c',
+               'import sys; from ipalib import facts; sys.exit(0 '
+               'if facts.is_ipa_configured() else 1);']
+
+        # Use new method which should return 0 since successful install
+        self.master.run_command(cmd)
```

Comment:
In this case, we want to be sure that `is_ipa_configured` correctly detects the new state (`installation.complete`), but at this point, it's an unknown state. I suggest making sure that `sstore.get_state("installation", "complete") is True`.

Reply:
The intention was to never return None anymore so it's complete or it isn't. I'll have to remove the is None comparisions for this.

Reply:
Subsequent tests should confirm that sstore is updated. I wanted to specifically test that the code in the spec file worked as it used to.

Reply:
I mean that the initial state of `installation.complete` is unknown (it may be missing at all). In this case, the positive scenario of the old method will be checked twice.

method | scenario | checked
-|-|-
old | positive | <ul><li>           [x] </li></ul>
old | negative | <ul><li>           [ ] </li></ul>
new | positive |               ?
new | negative | <ul><li>           [x] </li></ul>